src/ActionModal/components/ActionModal.vue

```diff
@@ -682,9 +685,17 @@ export default {
 
       // limit fees to the maximum the user has
       if (this.invoiceTotal > this.selectedBalance.amount) {
+        let payable = Number(this.subTotal)
+        // in terra we also have to pay the tax
+        // TODO refactor using a `fixedFee` property
```

Comment:
Where should we include this property? Inside the network JSON or better in balances, inside gasPrice I presume? And then gasPrice wouldn't be gasPrice anymore and would look like this:

```
fees: {
gasPrice: String,
fixedFee: String
}
```

Or what where your thoughts here?

Reply:
I don't now yet how to properly architect this. I would probably do al the calculation server side and just return a final fees object. Probably not handling all the edge cases client side. Just showing a warning: Your balance is not enough to pay for the transaction and fees.

Reply:
Yes, makes sense. We don't need to be so accurate. Just a warning/error message would be enough, I think too.

But we need to refactor a lot to do it like you suggest. Get simulate() and estimatedGasFee all those in the API. But sounds like a good approach.